broadinstitute/single_cell_portal_core
# Add timeout on CSFV (SCP-4035)

Adds a timeout to client side validation of 10 seconds. So if a user attempts to upload a super big file the CSFV will timeout after 10 seconds and print out a warning to the user letting them know that the validation will have to occur after file upload. Struggled to find good data on what the timeout time limit ought to be, went with 10 seconds as that felt acceptable anecdotally but again couldn't find any standards on this so input is welcome.

This adds a status type of 'timeout' for the mixpanel event as this isn't really a true "failure" or "success" and is presented as a "warning" to users in the UI.

also updates the iterateLines() function to take its params as an object for cleaner handling.
<img width="1212" alt="Screen Shot 2022-02-09 at 3 36 22 PM" src="https://user-images.githubusercontent.com/54322292/153285821-e1a7f495-0c9c-4753-9768-e55d0d63831a.png">

To test:
- pull this branch
- attempt to upload a super big file, such as the `MIT_Milk_Study_Raw_counts.txt` file which you can obtain from the study SCP1671 in prod
- observe that after 10 seconds the file stops trying to validate and a warning appears and you can click save
